modin/test/test_utils.py

```diff
@@ -287,3 +290,48 @@ def test_warns_that_defaulting_to_pandas():
 
     with warns_that_defaulting_to_pandas():
         ErrorMessage.default_to_pandas(message="Function name")
+
+
+def test_assert_dtypes_equal():
+    """Verify that `assert_dtypes_equal` from test utils works correctly (raises an error when it has to)."""
+    from modin.pandas.test.utils import assert_dtypes_equal
+
+    # Serieses with equal dtypes
+    sr1, sr2 = pd.Series([1.0]), pandas.Series([1.0])
+    assert sr1.dtype == sr2.dtype == "float"
+    assert_dtypes_equal(sr1, sr2)  # shouldn't raise an error since dtypes are equal
+
+    # Serieses with different dtypes belonging to the same class
+    sr1 = sr1.astype("int")
+    assert sr1.dtype != sr2.dtype and sr1.dtype == "int"
+    assert_dtypes_equal(sr1, sr2)  # shouldn't raise an error since both are numeric
```

Comment:
Why should we allow this? The dtypes are different.

Reply:
Currently we deviate from pandas significantly in terms of dtypes (even with this light requirements for types equality I've raised more than 15 issues when making this PR)

So in my opinion such requirements are quite reasonable as a first step

Reply:
Please create an issue to make more strong dtypes checking.